lib/spring/generate/springInitializrGenerator.ts

```diff
@@ -14,26 +14,46 @@
  * limitations under the License.
  */
 
+import { RemoteLocator } from "@atomist/automation-client/operations/common/params/RemoteLocator";
+import { RepoCreationParameters } from "@atomist/automation-client/operations/generate/RepoCreationParameters";
+import {
+    SmartParameters,
+    ValidationResult,
+} from "@atomist/automation-client/SmartParameters";
 import { doWithRetry } from "@atomist/automation-client/util/retry";
-import { NodeFsLocalProject, ParametersObject, Project, SoftwareDeliveryMachine } from "@atomist/sdm";
+import {
+    NodeFsLocalProject,
+    Parameter,
+    Parameters,
+    Project,
+    SoftwareDeliveryMachine,
+} from "@atomist/sdm";
 import axios from "axios";
 import * as decompress from "decompress";
 import * as fs from "fs";
 import * as path from "path";
 import * as tmp from "tmp-promise";
+import {
+    JavaIdentifierRegExp,
+    JavaPackageRegExp,
+    MavenArtifactIdRegExp,
+    MavenGroupIdRegExp,
+} from "../../java/javaPatterns";
 import { SetAtomistTeamInApplicationYml } from "./springBootTransforms";
 import {
-    SpringProjectCreationParameterDefinitions,
     SpringProjectCreationParameters,
 } from "./SpringProjectCreationParameters";
 import { TransformSeedToCustomProject } from "./transformSeedToCustomProject";
 
+let metaData: any;
+
 export function addSpringInitializrGenerator(sdm: SoftwareDeliveryMachine) {
+    getSpringInitializrMetaData();
```

Comment:
Can this be done in a SpringInitializrProjectCreationParameters constructor and stored as instance data on the object rather than a "global" value?

Reply:
done